README.md

```diff
@@ -320,6 +319,167 @@ override func prepareForSegue(_ segue: UIStoryboardSegue, sender sender: AnyObje
+```
+
+This will generate an `enum` for each of your `NSStoryboard`, with one `case` per storyboard scene.
+
+### Generated code
+
+The generated code will look like this:
+
+```swift
+import Foundation
+import Cocoa
+
+protocol StoryboardSceneType {
+  static var storyboardName: String { get }
+}
+
+extension StoryboardSceneType {
+  static func storyboard() -> NSStoryboard {
+    return NSStoryboard(name: self.storyboardName, bundle: nil)
+  }
+
+  static func initialController() -> AnyObject {
+    guard let controller = storyboard().instantiateInitialController()
+    else {
+      fatalError("Failed to instantiate initialViewController for \(self.storyboardName)")
+    }
+    return controller
+  }
+}
+
+extension StoryboardSceneType where Self: RawRepresentable, Self.RawValue == String {
+  func controller() -> AnyObject {
+    return Self.storyboard().instantiateControllerWithIdentifier(self.rawValue)
+  }
+  static func controller(identifier: Self) -> AnyObject {
+    return identifier.controller()
+  }
+}
+
+protocol StoryboardSegueType: RawRepresentable { }
+
+extension NSWindowController {
+  func performSegue<S: StoryboardSegueType where S.RawValue == String>(segue: S, sender: AnyObject? = nil) {
+    performSegueWithIdentifier(segue.rawValue, sender: sender)
+  }
+}
+
+extension NSViewController {
+  func performSegue<S: StoryboardSegueType where S.RawValue == String>(segue: S, sender: AnyObject? = nil) {
+    performSegueWithIdentifier(segue.rawValue, sender: sender)
+  }
+}
+
+struct StoryboardScene {
+  enum Anonymous_Osx: StoryboardSceneType {
+    static let storyboardName = "Anonymous-osx"
+  }
+  enum Dependency_Osx: String, StoryboardSceneType {
+    static let storyboardName = "Dependency-osx"
+
+    case DependentScene = "Dependent"
+    static func instantiateDependent() -> NSViewController {
+      guard let vc = StoryboardScene.Dependency_Osx.DependentScene.controller() as? NSViewController
+      else {
+        fatalError("ViewController 'Dependent' is not of the expected class NSViewController.")
+      }
+      return vc
+    }
+  }
+  enum Message_Osx: String, StoryboardSceneType {
+    static let storyboardName = "Message-osx"
+
+    case MessageDetailsScene = "MessageDetails"
+    static func instantiateMessageDetails() -> NSViewController {
+      guard let vc = StoryboardScene.Message_Osx.MessageDetailsScene.controller() as? NSViewController
+      else {
+        fatalError("ViewController 'MessageDetails' is not of the expected class NSViewController.")
+      }
+      return vc
+    }
+
+    case MessageListScene = "MessageList"
+    static func instantiateMessageList() -> NSViewController {
+      guard let vc = StoryboardScene.Message_Osx.MessageListScene.controller() as? NSViewController
+      else {
+        fatalError("ViewController 'MessageList' is not of the expected class NSViewController.")
+      }
+      return vc
+    }
+
+    case MessagesTabScene = "MessagesTab"
```

Comment:
This example / sample code is probably a bit too long for a README. Maybe add a `[… snip …]` in the middle to truncate it, or simply remove some cases to only show 1 or 2 in the README. After all, it's there only to show the general principle of how the generated code will look like, to give the new user an overlook, not to detail every use case :wink: